engine/api/workflow/execute_node_run.go

```diff
@@ -332,12 +332,8 @@ func addJobsToQueue(ctx context.Context, db gorp.SqlExecutor, stage *sdk.Stage,
 	report := new(ProcessorReport)
 
 	_, next := observability.Span(ctx, "sdk.WorkflowCheckConditions")
-	conditionsOK, err := sdk.WorkflowCheckConditions(stage.Conditions(), run.BuildParameters)
+	conditionsOK := checkNodeRunCondition(wr, stage.Conditions, run.BuildParameters)
```

Comment:
Can we rename this function to checkCondition, because it doesn't check node run condition but also stages now ?
and change the observability span name